kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NMDbusConnectorTest.java

```diff
@@ -0,0 +1,619 @@
+    NMDbusConnector instanceNMDbusConnector;
+    DBusConnection dbusConnectionInternal;
+
+    Boolean hasDBusExceptionBeenThrown = false;
+    Boolean hasNoSuchElementExceptionThrown = false;
+    Boolean hasNullPointerExceptionThrown = false;
+
+    NetworkInterfaceStatus netInterface;
+    NetworkService networkService;
+
+    Map<String, Device> mockDevices = new HashMap<>();
+
+    List<String> internalStringList;
+    Map<String, Object> netConfig = new HashMap<>();
+
+    @After
+    public void tearDown() {
+        resetSingleton(NMDbusConnector.class, "instance");
+    }
+
+    public static <T> void resetSingleton(Class<T> clazz, String fieldName) {
+        Field instance;
+        try {
+            instance = clazz.getDeclaredField(fieldName);
+            instance.setAccessible(true);
+            instance.set(null, null);
+            instance.setAccessible(false);
+        } catch (Exception e) {
+            throw new RuntimeException();
+        }
+    }
+
+    @Test
+    public void getDbusConnectionShouldWork() throws DBusException {
+        givenBasicMockedDbusConnector();
+        whenGetDbusConnectionIsRun();
+
+        thenNoExceptionIsThrown();
+
+        thenGetDbusConnectionIsMockedConnection();
+    }
+
+    @Test
+    public void checkPermissionsShouldWork() throws DBusException {
+        givenBasicMockedDbusConnector();
+        givenMockedPermissions();
+
+        whenCheckPermissionsIsRun();
+
+        thenNoExceptionIsThrown();
+        thenCheckPermissionsRan();
+    }
+
+    @Test
+    public void checkVersionShouldWork() throws DBusException {
+        givenBasicMockedDbusConnector();
+        givenMockedVersion();
+
+        whenCheckVersionIsRun();
+
+        thenNoExceptionIsThrown();
+        thenCheckVersionIsRun();
+    }
+
+    @Test
+    public void getInterfacesShouldWork() throws DBusException {
+        givenBasicMockedDbusConnector();
+        givenMockedDevice("eth0", NMDeviceType.NM_DEVICE_TYPE_ETHERNET, NMDeviceState.NM_DEVICE_STATE_DISCONNECTED,
+                true);
+        givenMockedDevice("wlan0", NMDeviceType.NM_DEVICE_TYPE_WIFI, NMDeviceState.NM_DEVICE_STATE_DISCONNECTED, true);
+        givenMockedDeviceList();
+
+        whenGetInterfacesIsCalled();
+
+        thenNoExceptionIsThrown();
+        thenReturnedDevicesAre(Arrays.asList("wlan0", "eth0"));
+    }
+
+    @Test
+    public void applyShouldDoNothingWithNoCache() throws DBusException {
+        givenBasicMockedDbusConnector();
+        givenMockedDevice("eth0", NMDeviceType.NM_DEVICE_TYPE_ETHERNET, NMDeviceState.NM_DEVICE_STATE_DISCONNECTED,
+                true);
+        givenMockedDevice("wlan0", NMDeviceType.NM_DEVICE_TYPE_WIFI, NMDeviceState.NM_DEVICE_STATE_DISCONNECTED, true);
+        givenMockedDeviceList();
+
+        whenApplyIsCalled();
+
+        thenNoExceptionIsThrown();
+        thenNetworkSettingsDoNotChangeForDevice("eth0");
+        thenNetworkSettingsDoNotChangeForDevice("wlan0");
```

Comment:
Consistency: the other methods use the past tense. Use either the present tense or the past tense not both.

```suggestion
        thenNetworkSettingsDidNotChangeForDevice("eth0");
        thenNetworkSettingsDidNotChangeForDevice("wlan0");
```